e2e/_suites/ingest-manager/features/agent_endpoint_integration.feature

```diff
@@ -0,0 +1,38 @@
+@agent_endpoint_integration
+Feature: Fleet Mode Agent with Endpoint Integration
+  Scenarios for Agent to deploy Endpoint and sending data to Ingest Manager and Elasticsearch.
+
+@deploy-endpoint-with-agent
+Scenario: Deploy Agent and add Endpoint Integration and host shows in Security App
+  Given an agent is deployed to Fleet
+  When the "Endpoint" latest package version is installed successfully
+    And the "Endpoint" integration is "added" in the "default" configuration
+  Then the "default" configuration shows the "Endpoint" datasource added
+    And the host name is shown in the security app
+    
+@endpoint-policy-check
+Scenario: Deploy Endpoint and see policy return successful in Security App
+  Given an Endpoint is successfully deployed with Agent
```

Comment:
What is an Endpoint in terms of Docker image? Is it a Centos/Debian image? I ask because the test framework needs a representation of the box in the form of a Docker container

Reply:
Related to this, what are the implications of "deploying and Endpoint with Agent"? Does it mean installing the agent in the box representing the Endpoint?

Reply:
Further, my intention was to re-use the implementation code bundled together from the above individual steps, called from the one Given listed here as a short hand for what all of the tests are doing for setup.

so...

Given an Endpoint is successfully deployed with Agent

is basically equivalent to:

  When the "Endpoint" latest package version is installed successfully
    And the "Endpoint" integration is "added" in the "default" configuration
  Then the "default" configuration shows the "Endpoint" datasource added
    And the host name is shown in the Security App